.github/workflows/node-ci.yml

```diff
@@ -143,17 +116,22 @@ jobs:
 
   # This workflow will build distributable artifacts and commit them to the
   # branch that is being used for the PR. This should work even on protected
-  # branches because of our use of an SSH deploy key that is setup against
-  # the repo.
+  # branches because of secrets. This also is responsible for running and
+  # uploading code coverage since it already runs the tests before the build
+  # anyway.
   publish_to_branch:
-    needs: [prime_cache]
-    name: Publish to branch
+    needs: [prime_cache_primary]
+    name: Publish to branch and gather coverage
     # We only want to run this if it's a push event or a pull_request for
     # a non-feature branch.
     if: github.event_name == 'push' || !startsWith(github.head_ref, 'feature/')
     env:
       CI: true
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node-version: [12.x]
```

Comment:
Changed to using a matrix to standardize the yaml and also to make the task names more self-documenting in the Actions summaries.